test/test_torch.py

```diff
@@ -2644,6 +2644,18 @@ def test_copy_dtypes(self):
             copied_dtype = copy.deepcopy(dtype)
             self.assertIs(dtype, copied_dtype)
 
+    def test_copy_transpose(self):
+        x = torch.arange(100 * 100, dtype=torch.float).reshape(100, 100).t()
+        y = torch.empty(100, 100, dtype=torch.float)
+        y.copy_(x)
+        self.assertEqual(y[:, 0], range(100))
+        self.assertEqual(y[:, 40], range(4000, 4100))
+
+        y = torch.empty(100, 100, dtype=torch.double)
+        y.copy_(x)
```

Comment:
This tests copy from a transposed float array (x) to a contiguous float/double array (y). Though in my example it's double->float direction that breaks. Not sure it changes much though

Reply:
The test captures the aspect that it's copying between different data types, which is all that matters here.